### ISSUE TITLE
1.38 Doesn't work with ReHLDS + ReGameDLL_CS ?

### ISSUE THREAD

**Issue by Freeman-AM** [jkivilin/metamod-p#10]
tmux console log

```
Auto-restarting the server on crash                                                                                                                                [7/7]

Console initialized.
Using breakpad crash handler
Setting breakpad minidump AppID = 10
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
Protocol version 48
Exe version 1.1.2.7/Stdio (cstrike)
Exe build: 15:11:44 Apr  1 2018 (1450)
STEAM Auth Server
Server logging data to file logs/L0405005.log
L 04/05/2018 - 19:13:22: Log file started (file "logs/L0405005.log") (game "cstrike") (version "48/1.1.2.7/Stdio/1450")
Server IP address 164.132.202.76:27015

   Metamod version 1.21p38 Copyright (c) 2001-2013 Will Day
     Patch: Metamod-P (mm-p) v38 Copyright (c) 2004-2018 Jussi Kivilinna
   Metamod comes with ABSOLUTELY NO WARRANTY; for details type `meta gpl'.
   This is free software, and you are welcome to redistribute it
   under certain conditions; type `meta gpl' for details.

L 04/05/2018 - 19:13:22: [META] Metamod v1.21p38  2018/02/11
L 04/05/2018 - 19:13:22: [META] by Will Day
L 04/05/2018 - 19:13:22: [META]    http://www.metamod.org/
L 04/05/2018 - 19:13:22: [META]  Patch: Metamod-P (mm-p) v38
L 04/05/2018 - 19:13:22: [META]  by Jussi Kivilinna
L 04/05/2018 - 19:13:22: [META]     http://metamod-p.sourceforge.net/
L 04/05/2018 - 19:13:22: [META] compiled: Feb 11 2018, 11:05:06 EET (optimized)
L 04/05/2018 - 19:13:22: [META] WARNING: GameDLL-Autodetection: Couldn't find gamedll in '/home/server1/ffa/cstrike/dlls'.
L 04/05/2018 - 19:13:22: [META] Recognized game 'cstrike'; using dllfile 'cs_amd64.so'
L 04/05/2018 - 19:13:22: [META] WARNING: dll: Couldn't load game DLL /home/server1/ffa/cstrike/dlls//home/server1/ffa/cstrike/dlls/cs_amd64.so: /home/server1/ffa/cstrik
e/dlls//home/server1/ffa/cstrike/dlls/cs_amd64.so: cannot open shared object file: No such file or directory
L 04/05/2018 - 19:13:22: [META] ERROR: Failure to load game DLL; exiting...
L 04/05/2018 - 19:13:22: [META] WARNING: GetEntityAPI2 called with null pFunctionTable
==================
Game DLL version mismatch
DLL version is 140, engine version is 140
The game DLL for cstrike appears to be outdated, check for updates
==================
Host_Error:

L 04/05/2018 - 19:15:22: FATAL ERROR (shutting down): Host_Error:


FATAL ERROR (shutting down): Host_Error:


./hlds_run: line 255: 12419 Segmentation fault      $HL_CMD
```

sys_error.log :

> L 05/04/2018 - 19:17:16: (map "(null)") Host_Error: 

It's working fine when i rollback to 1.37

**Comment by Freeman-AM** [comment]
It's working fine when i roll back to 1.37

**Comment by jkivilin** [comment]
> L 04/05/2018 - 19:13:22: [META] WARNING: dll: Couldn't load game DLL /home/server1/ffa/cstrike/dlls//home/server1/ffa/cstrike/dlls/cs_amd64.so: /home/server1/ffa/cstrik
e/dlls//home/server1/ffa/cstrike/dlls/cs_amd64.so: cannot open shared object file: No such file or directory

Search path for game DLL seems a bit strange.

**Comment by Freeman-AM** [comment]
Well i followed the default ReHLDS/ReGameDLL installation

my liblist.gam

```
game "Counter-Strike"
url_info "www.counter-strike.net"
url_dl ""
version "1.6"
size "184000000"
svonly "0"
secure "1"
type "multiplayer_only"
cldll "1" 
hlversion "1111"
nomodels "1"
nohimodel "1"
mpentity "info_player_start"
gamedll "dlls\mp.dll"
gamedll_linux "addons/metamod/dlls/metamod.so"
gamedll_osx "dlls/cs.dylib"
trainmap "tr_1"
edicts	"1800"
```

I'm running on Debian 64 bits.

**Comment by Frisasky** [comment]
if you re running rehlds please use this version
https://github.com/theAsmodai/metamod-r
the metamod-p only works with vanilla hlds

**Comment by Freeman-AM** [comment]
Why closure @jkivilin ?  

@Frisasky >> the metamod-p only works with vanilla hlds

False, stop saying stupid things when you don't know the topic.

Metamod-p is even recommended for the installation of rehlds regamedll_CS
metamod-p work fine even the new version.
It's just the automatic detection of game path that is broken with the new version...

I was asking for a fix of that feature.

metamod-r is higly unstable, it's not recommended in a "production" environnement server.

**Comment by germansassone** [comment]
@Freeman-AM
_________________________________________
Try this:
In addons/metamod create the file config.ini
In config.ini edit the file and put gamedll dlls/cs.so
_________________________________________
I can confirm that the combination rehlds, regamedll, metamod-p is working just fine by now
_________________________________________

**Comment by APGRoboCop** [comment]
Well apart from reHLDS and ReGameDLL I think I've just experienced that for HL Science & Industry as well as Op4, CS 1.6 and CZero. I had to go back to using Metamod-P37 for that problem with the that error:

> ==================
> Game DLL version mismatch
> DLL version is 140, engine version is 140
> The game DLL for cstrike appears to be outdated, check for updates
> ==================

Forget using Metamod-R as they are unstable and crashes a lot.

**Comment by jkivilin** [comment]
The "Game DLL version mismatch / DLL version is 140, engine version is 140" message comes from the **engine itself**, not from metamod.

When metamod fails to load the game DLL (e.g. wrong SO filename, path issues), the engine falls back to its own loading code. In ReHLDS `engine/sys_dll.cpp:1000-1011`, when `GetEntityAPI2` returns false, the engine unconditionally prints the "Game DLL version mismatch" block — regardless of whether the versions actually differ. The version check only controls which "outdated" flavour to print, not whether to print it at all. This is an engine bug.

The actual metamod-side problems that caused the game DLL load failures have been addressed:
- Missing/outdated SO names in `games.h` — fixed by #59 and #62
- Workaround: specify `gamedll dlls/<name>.so` in `config.ini` (as noted in comments)
